docs/coding-guidelines.md

```diff
@@ -333,6 +333,55 @@ The person writing the code will write the doc comments. Public APIs only. No ne
 
 Note: Public means callable by a customer, so it includes protected APIs. However, some public APIs might still be "for internal use only" but need to be public for technical reasons. We will still have doc comments for these APIs but they will be documented as appropriate.
 
+- Do not include empty XML comments.
+  - summary element must not be empty.
+  - param element must not be empty.
+
+Correct:
+
+```cs
+        /// <summary>
+        /// Calculates all fruit types
+        /// </summary>
+        /// <returns>A set of fruit types contained in this basket.</returns>
+        public ISet<FruitType> GetAllFruitTypes()
+
+        /// <summary>
+        /// Determines whether the fruit has seeds.
+        /// </summary>
+        /// <param name="fruitType">A fruit type</param>
+        /// <returns> Whether the fruit has seeds. </returns>
+        public bool HasSeeds(FruitType fruitType)
+        
+        /// <summary>
+        /// Determines whether the fruit is multiannual
+        /// </summary>
+        /// <returns> Whether the fruit is multiannual. </returns>
```

Comment:
Why do we want to encourage this or say it's correct? I believe other comments from zivkan and myself point toward this as bad practice.

Reply:
I'm suggesting *all params* or *no params*. 

It's the closest to how our current APIs are documented.

Reply:
For new guidance, I'd prefer we always include all params, even if just a boilerplate string is used in the content.
However, it's just my preference and I won't  block on it.

Reply:
> However, it's just my preference and I won't block on it.

Let's get a 3rd opinion. 

@zivkan or someone else from @NuGet/nuget-client 

I'm happy to go either direction.

Reply:
If we can enforce it at a tooling level, I agree with Donnie that all **public** methods should have **all** params documented, plus the `return` if non-`void`.

I don't think we should waste time in code reviews over this, though.